File: ChainJava/src/chain/employees/ClubPresident.java
package chain.employees;

import chain.requests.Request;

/**
 * Created by Karolina_2 on 2015-04-01.
 */
public class ClubPresident extends Employee {

    public ClubPresident(){

    }

    @Override
    public void receiveARequest(Request request){
        if(isAllowedToReceive(request, "ContractRequest")){
            System.out.println("Club president has received " + request.getClass().getSimpleName());
        }
        else{
            super.receiveARequest(request);
        }
    }
}

File: Chain/Chain/chain/employees/ClubPresident.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ZTP_Zadanie3.chain.wnioski;

namespace ZTP_Zadanie3.pracownicy
{
    class ClubPresident : Employee
    {
        public ClubPresident() { }

        public override void receiveARequest(Request request)
        {
            if (isAllowedToReceive(request, "ContractRequest"))
            {
                Console.WriteLine("Club president has received " + request.GetType().Name);
            }
            else
            {
                base.receiveARequest(request);
            }
        }
    }
}
